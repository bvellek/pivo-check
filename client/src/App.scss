@import "scss/vendor/_normalize";


@import "scss/base/_type";
@import "scss/base/_colors";
@import "scss/base/_size-variables";
@import "scss/base/_general";
@import "scss/base/_visually-hidden";



@import "scss/components/_no-js";
@import "scss/components/_header";
@import "scss/components/_landing-page";
@import "scss/components/_loader";
@import "scss/components/_footer";
@import "scss/components/_login-page";
@import "scss/components/_registration-page";
@import "scss/components/_cities-page";
@import "scss/components/_city-page";
@import "scss/components/_error-message";
