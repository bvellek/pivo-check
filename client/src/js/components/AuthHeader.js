import React from 'react';
import { Link } from 'react-router-dom';
import SVGLogo from './SVGLogo';

const AuthHeader = () => (
  <header className="auth-header top-header" role="banner">
    <Link to='/'>
      <h1>
        <span className="visually-hidden">Pivo-Check</span>
        <SVGLogo />
      </h1>
    </Link>
  </header>
);

export default AuthHeader;
