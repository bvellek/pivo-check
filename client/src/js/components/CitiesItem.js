import React from 'react';
import { Link } from 'react-router-dom';
import SVGDelete from './SVGDelete';

const CitiesItem = ({
  cityInfo,
}) => (
  <div>
    <div className="city-item">
      <Link to={`/cities/${cityInfo._id}`}><h2 className="city-list-title">{cityInfo.cityName}</h2></Link>
      <div className="city-info">
        <p className="city-list-brew-count">{cityInfo.completedCount}/{cityInfo.brewTotal}</p>
        <form className="city-list-delete-form" action="index.html" method="post">
          <button type="submit" name="button"><SVGDelete /><span className="visually-hidden">Delete {cityInfo.cityName}</span></button>
        </form>
      </div>
    </div>

    <div className="delete-city-warning">
      <h3>Delete {cityInfo.cityName}?</h3>
      <p>Deleting a location will permanently remove it from your checklist and you will lose your visited brewery&nbsp;record.</p>
      <form className="delete-city-form" action="index.html" method="post">
        <button className="keep-city-btn" type="button" name="button">No, Keep {cityInfo.cityName}</button> {' '}
        <button className="delete-city-btn" type="button" name="button"><SVGDelete />&nbsp;Yes, Delete {cityInfo.cityName}</button>
      </form>
    </div>
  </div>
);

export default CitiesItem;
