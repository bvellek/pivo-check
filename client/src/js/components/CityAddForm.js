import React, { Component } from 'react';
import { connect } from 'react-redux';
import PlacesAutocomplete, { geocodeByAddress } from 'react-places-autocomplete';
import * as actions from '../actions/index';

export class CityAddForm extends Component {
  constructor(props) {
    super(props);
    this.state = { address: '' };
    this.onChange = (address) => this.setState({ address });
  }

  handleFormSubmit = (event) => {
    event.preventDefault();
    const { address } = this.state;

    geocodeByAddress(address, (err, { lat, lng }, results) => {
      if (err) { console.log('Oh no!', err); } // eslint-disable-line
      // the State name is not always returned in the same array item, so filter by the type
      const filterStateResults = results[0].address_components.filter((object) => (
        object.types[0] === 'administrative_area_level_1'
      ));
      this.props.dispatch(actions.addCity({
        coords: {
          lat,
          lng,
        },
        cityName: `${results[0].address_components[0].long_name}, ${filterStateResults[0].short_name}`,
      }));
    });
  }

  handleSelect = (address) => {
    this.setState({ address });

    geocodeByAddress(address, (err, { lat, lng }, results) => {
      if (err) { console.log('Oh no!', err); } // eslint-disable-line
      // the State name is not always returned in the same array item, so filter by the type
      const filterStateResults = results[0].address_components.filter((object) => (
        object.types[0] === 'administrative_area_level_1'
      ));
      this.props.dispatch(actions.addCity({
        coords: {
          lat,
          lng,
        },
        cityName: `${results[0].address_components[0].long_name}, ${filterStateResults[0].short_name}`,
      }));
    });
  }

  render() {
    const options = {
      componentRestrictions: { country: 'US' },
      types: ['(cities)'],
    };

    const myStyles = {
      root: {
        position: 'relative',
        margin: '0 auto',
        width: '100%',
        maxWidth: '22em',
        padding: '0',
      },
      input: {
        boxSizing: 'border-box',
        width: '100%',
        lineHeight: '2em',
        margin: '0 auto',
        padding: '0 0.2em',
        color: '#44484D',
      },
      autocompleteItemActive: {
        backgroundColor: '#FFBAB3',
      },
    };

    const AutocompleteItem = ({ formattedSuggestion }) => (
      <div>
        <strong>{ formattedSuggestion.mainText }</strong>{' '}
        <small>{ formattedSuggestion.secondaryText }</small>
      </div>
    );

    return (

      <section className='city-add'>
        <form
          className='city-add-form'
          method='post'
          autoComplete='off'
          onSubmit={this.handleFormSubmit}
        >
          <div className='city-form-content'>
            <label htmlFor='city-search'>Add a city</label>
            <PlacesAutocomplete
              value={this.state.address}
              onChange={this.onChange}
              onSelect={this.handleSelect}
              options={options}
              inputId={'city-search'}
              inputName={'city'}
              styles={myStyles}
              autocompleteItem={AutocompleteItem}
              placeholder={'Seattle, WA'}
            />
            <button aria-label='Add city' type='submit' name='addCity'>Submit City 📍</button>
          </div>
        </form>
      </section>
    );
  }
}

export default connect()(CityAddForm);

