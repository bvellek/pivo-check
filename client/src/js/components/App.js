import React, { Component } from 'react';
import '../../App.css';
import LandingHeader from './LandingHeader';
import LandingPage from './LandingPage';

class App extends Component {
  render() {
    return (
      <div className="App">
        <LandingHeader />
        <LandingPage />
      </div>
    );
  }
}

export default App;
