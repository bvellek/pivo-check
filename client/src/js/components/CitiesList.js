import React from 'react';
import CitiesItem from './CitiesItem';

const CitiesList = ({
  cities,
}) => (
  <section className="cities-list">
    <ul>
      {cities.map((city, index) => (
        <li key={index}><CitiesItem cityInfo={city} /></li>
      ))}
    </ul>
  </section>
);

export default CitiesList;
