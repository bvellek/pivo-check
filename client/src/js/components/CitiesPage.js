import React, { Component } from 'react';
import { connect } from 'react-redux';
import * as actions from '../actions/index';
import InAppHeader from './InAppHeader';
import CityAddForm from './CityAddForm';
import SVGDelete from './SVGDelete';
import Footer from './Footer';

class CitiesPage extends Component {
  constructor(props) {
    super(props);
    this.state = {
    };
  }

  componentDidMount() {
    this.props.dispatch(actions.cleanAuth());
  }

  render() {
    return (
      <div>
        <InAppHeader />
        <div className="cities-page">
          <main role="main">
            <h1>My Cities</h1>
            <section className="city-add">
              <CityAddForm />
            </section>

          {/* if no cities added */}
            <section className="no-cities">
              <h2>Add a City!</h2>
              <p>Looks like you haven't added any cities to your checklist. Submit above to add a city full of breweries!</p>
            </section>

        {/* default */}
            <section className="cities-list">
              <ul>
                <li>
                  <div className="city-item">
                    <a href="#"><h2 className="city-list-title">Seattle, WA</h2></a>
                    <div className="city-info">
                      <p className="city-list-brew-count">7/20</p>
                      <form className="city-list-delete-form" action="index.html" method="post">
                        <button type="submit" name="button"><SVGDelete /><span className="visually-hidden">Delete Seattle</span></button>
                      </form>
                    </div>
                  </div>

                  <div className="delete-city-warning">
                    <h3>Delete Seattle?</h3>
                    <p>Deleting a location will permanently remove it from your checklist and you will lose your visited brewery record.</p>
                    <form className="delete-city-form" action="index.html" method="post">
                      <button className="keep-city-btn" type="button" name="button">No, Keep Seattle</button> {' '}
                      <button className="delete-city-btn" type="button" name="button"><SVGDelete />&nbsp;Yes, Delete Seattle</button>
                    </form>
                  </div>
                </li>

                <li>
                  <div className="city-item">
                    <a href="#"><h2 className="city-list-title">Portland, OR</h2></a>
                    <div className="city-info">
                      <p className="city-list-brew-count">2/68</p>
                      <form className="city-list-delete-form" action="index.html" method="post">
                        <button type="button" name="button"><SVGDelete /><span className="visually-hidden">Delete Portland</span></button>
                      </form>
                    </div>
                  </div>
                </li>
                <li>
                  <div className="city-item">
                    <a href="#"><h2 className="city-list-title">FL. Pops Hammock Seminole Village, FL</h2></a>
                    <div className="city-info">
                      <p className="city-list-brew-count">5/19</p>
                      <form className="city-list-delete-form" action="index.html" method="post">
                        <button type="button" name="button"><SVGDelete /><span className="visually-hidden">Delete Austin</span></button>
                      </form>
                    </div>
                  </div>
                </li>
              </ul>
            </section>
          </main>
          <Footer />
        </div>
      </div>
    );
  }
}

export default connect()(CitiesPage);

