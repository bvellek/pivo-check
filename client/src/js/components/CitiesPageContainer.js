import React, { Component } from 'react';
import { connect } from 'react-redux';
import * as actions from '../actions/index';
import CitiesPage from './CitiesPage';
import CitiesList from './CitiesList';
import NoCities from './NoCities';

class CitiesPageContainer extends Component {
  constructor(props) {
    super(props);

    this.state = {
    };
  }

  componentDidMount() {
    document.querySelector('head > title').innerHTML = 'My Cities | PIVO-CHECK';
    this.props.dispatch(actions.cleanAuth());
    this.props.dispatch(actions.getCities());
  }

  addCity = (e) => {
    e.preventDefault();
  }

  // changeUser = (e) => {
  //   const field = e.target.name;
  //   const user = this.state.user;
  //   user[field] = e.target.value;

  //   this.setState({
  //     user,
  //   });
  // }

  // processForm = (e) => {
  //   e.preventDefault();
  //   this.props.dispatch(actions.loginUser(this.state.user));
  //   console.log(`email: ${this.state.user.email}`);
  //   console.log(`password: ${this.state.user.password}`);
  // }

  // demoSubmit = (e) => {
  //   e.preventDefault();
  //   this.setState({
  //     user: {
  //       password: 'demoPassword',
  //       email: 'demouser@PivoCheck.com',
  //     },
  //   }, () => {
  //     document.querySelector('.login-btn').click();
  //   });
  //   console.log(`email: ${this.state.user.email}`);
  //   console.log(`password: ${this.state.user.password}`);
  // }

  render() {
    const citiesArr = this.props.myCities;
    let citiesDisplay;
    if (citiesArr.length > 0) {
      citiesDisplay = <CitiesList cities={citiesArr} />;
    } else {
      citiesDisplay = <NoCities />;
    }

    return (
      <CitiesPage
        loadingStatus={this.props.loadingStatus}
        addCity={this.addCity}
        myCities={citiesDisplay}
      />
    );
  }
}

const mapStateToProps = (state) => ({
  loadingStatus: state.cities.citiesLoadingStatus,
  myCities: state.cities.myCities,
});

export default connect(mapStateToProps)(CitiesPageContainer);


// <Route exact path="/login" render={() => ( // eslint-disable-line
//           !this.props.loginRedirect
//           ? (<LoginPage
//             onSubmit={this.processForm}
//             onChange={this.changeUser}
//             demoSubmit={this.demoSubmit}
//             user={this.state.user}
//             registrationSuccessMessage={this.props.registrationSuccessMessage}
//             errors={this.props.errors}
//           />)
//           : <Redirect to="/cities" />
//         )}
//       />
