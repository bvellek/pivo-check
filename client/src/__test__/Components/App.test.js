import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';

import { App } from '../../js/components/App';

describe('<App />', () => {
  it('renders without crashing', () => {
    shallow(<App />);
  });
  it('it renders correctly', () => {
    const component = shallow(
      <App />
    );
    expect(shallowToJson(component)).toMatchSnapshot();
  });
});
