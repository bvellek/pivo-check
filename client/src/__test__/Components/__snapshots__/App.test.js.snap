exports[`<App /> it renders correctly 1`] = `
<BrowserRouter
  basename="/">
  <withRouter(ScrollToTop)>
    <div
      className="App">
      <Route
        exact={true}
        path="/"
        render={[Function]} />
      <Route
        path="/login"
        render={[Function]} />
      <Route
        path="/registration"
        render={[Function]} />
      <Route
        exact={true}
        path="/cities"
        render={[Function]} />
      <Route
        path="/cities/:cityID"
        render={[Function]} />
    </div>
  </withRouter(ScrollToTop)>
</BrowserRouter>
`;
